Fix the usability with multiple PLC interfaces and removed the dependency for old libgpiod1

s-hemer opened the pull request:
Ported some missing commits from kirkstone, did some rework of recipes and services. For the https://github.com/phytec/meta-lumissil-greenphy/tree/WIP/ggiordano@phytec.com/scarthgap_fix-build-errors commits, I did not see a need, it build fine/as expected for me.

There are still three things to be solved:
* on my custom HW, I have issues with the second iface but that seems to be due to SPI (hw) problems.
* building the host_loading_service example with release SDK (-> that is why last commit is marked with HACK).
* Squashing the 4.05 upgrade commits, kept them as they are for better traceability for now (just added some description).

dnltz committed:
recipes-utils: cg5317-utils: Use correct gpiochip

gpiochip1 represents GPIO0 located in the main domain.
gpiochip0 represents MCU_GPIO0 located in the MCU domain.

Use gpiochip1 as this is the correct gpiochip now after
MCU_GPIO0 got enabled.

Remove execution rights from service files.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>
Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

dnltz committed:
Move recipes-open-plc-utils into recipes-utils: open-plc-utils

Move the recipe since recipes-open-plc-utils is not a correct Yocto
sub-directory. Also add some patches to this tool.

Signed-off-by: Daniel Schultz <d.schultz@phytec.de>
(remove execution right from patch and ini files)
Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

alexatphytec committed:
recipes-utils: cg5317-utils: moved to version 04.05.000.0020

renamed the spi_*_configs, update firmware, update kernel docu, update
binary examples/tools

Also updated the moved ini files for open-plc-utils.

Signed-off-by: Alexander Bauer <a.bauer@phytec.de>
Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer committed:
recipes-utils: cg5317-utils: Fix systemd service install

This fix does install the services the correct way to prevent QA warning
due to symlinks pointing to TMPDIR.

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer committed:
recipes-utils: cg5317-utils: Replace bringup service with systemd
network file

Preparation for easier setup of multiple interfaces.

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer committed:
recipes-utils: cg5317-utils: Make FW service instantiable and add config

That allows easy re-use of the service itself for other interfaces.

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

s-hemer committed:
recipes-utils: cg5317-utils: Make firmware loading compatible with
libgpiod2

Fixed the dependency to the old libgpiod1 call for setting the reset
gpio (based on the v4.05 example code).
Build with release toolchain of PD24.2.1.

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

dnltz commented on recipes-utils/cg5317-utils/files/25-seth0.network:
```diff
@@ -0,0 +1,10 @@
+
+[Network]
+DHCP=yes
+Address=169.254.200.130/16
```
any reason why we hard-code an IP here when DHCP is enabled?

s-hemer replied:
That is a link-local address as fallback. I did this because the .network in the v4.05 SDK had it (but with 169.250.x.x, which is not link-local).

s-hemer replied:
Potentially you are right and this is not necessary (or `LinkLocalAddressing=ipv4` may do the same) and I was misled because the SDK used the x.250.x.x to prevent the link from being marked as degraded.

zwh2 commented on recipes-utils/cg5317-utils/cg5317-utils.bb:
Please do not rename these files. The customer needs to interact with them and changing the names will mean documentation needs to change.

s-hemer commented:
One more ToDo to think about: where to put the changes on SDK/source of host_loading_service binary

s-hemer committed:
recipes-utils: cg5317-utils: Install firmware and examples to subfolder

As with the config dir, separate the firmware and examples to a
'lumissil' specific subdir. Retore using the ROOT_HOME variable (without
re-setting it).

Signed-off-by: Steffen Hemer <s.hemer@phytec.de>

smk-embedded commented:
can we merge this and handle the todos in the next PR?

s-hemer commented:
In the meanwhile, it got quite some conflicts as @ggiordano32 pushed changes to scarthgap w/o reviewing this branch. I noticed that already some time ago, yet had no time to resolve. Just started that on friday again.

ggiordano32 committed:
conf: layer: drop libgpiod preference from layer

Use the scarthgap libgpiod as libgpiod.so.3 is deployed by default

Signed-off-by: Garrett Giordano <ggiordano@phytec.com>

ggiordano32 commented:
Rebased the changes here onto the current scarthgap branch. I included all commits:
https://github.com/phytec/meta-lumissil-greenphy/pull/6

smk-embedded commented:
Thanks @ggiordano32, if you agree @s-hemer and the solution works for you, can we move on, or are there other issues still open?

s-hemer commented:
It is quite a diff to #6, lot of minor cleanups, squashed to 04.05 SDK update ...
I only did build-test it for now. My HW is in secure-boot dev state (so no keys burnt yet), that takes a while to revert it to standard image and back to security again (loading provisioning image via tftp etc ...).
@ggiordano32 are you able to test easily? Then, I'd say, we can merge.

ggiordano32 commented:
I have hardware here, I will give it a test on my end and let you know @s-hemer

s-hemer commented:
Made one little mistake while installing the examples, fixed it. Firmware is loading, looks good so far from my side.

ggiordano32 commented:
@s-hemer that did it. Firmware loaded on my side. Great job!
